src/WorkBC.ElasticSearch.Indexing/Services/XmlParsingServiceFederal.cs

```diff
@@ -18,18 +18,19 @@ public class XmlParsingServiceFederal : XmlParsingServiceBase
         private const int BenefitsSkillCategoryId = 102001;
         private const string VirtualJobBasedIn_EN = "Virtual job based in";
         private const string VirtualJobBasedIn_FR = "Emploi virtuel basé à";
-        private readonly IGeocodingService _geocodingService;
+        private readonly IGeocodingService _geocodingCachingService;
 
 
         public XmlParsingServiceFederal(IConfiguration configuration) : base(configuration)
         {
-            _geocodingService = new GeocodingService(JobBoardContext, configuration);
+            var geoService = new GeocodingService(configuration);
+            _geocodingCachingService = new GeocodingCachingService(JobBoardContext, geoService);
         }
 
         public XmlParsingServiceFederal(List<Data.Model.JobBoard.Location> duplicateCities,
             Dictionary<string, string> uniqueCities, List<NocCode> nocCodes, IGeocodingService geocodingService) : base(duplicateCities, uniqueCities, nocCodes)
         {
-            this._geocodingService = geocodingService;
+            this._geocodingCachingService = geocodingService;
```

Comment:
I'm not sure I understand what's happening here.  The geocodingService in the dictionary needs to be an instance of the caching service and not the geocodingService that calls out to Google's API.

Reply:
Looks like a constructor made to copy one instance to another. We can just assume the incoming `geocodingService` is already a member attribute of another instance of `XmlParsingServiceFederal`.